machine-learning/db/QueryBuilder.py

```diff
@@ -190,16 +192,19 @@ def get_instance_fields(instance_name: str, fields, conditions: str = "", datase
     join_conditions = join_conditions[:-5]
 
     sql: str = "SELECT " + required_fields + " FROM " + required_tables + " WHERE " + join_conditions
```

Comment:
why are the joins modelled as WHERE conditions? Not sure if this has a performance impact...

Reply:
I will change this in another PR.